Source/DZNSegmentedControl.m

```diff
@@ -781,6 +784,8 @@ - (void)configureButtonForSegment:(NSUInteger)segment
 - (void)willSelectedButton:(id)sender
 {
     UIButton *button = (UIButton *)sender;
+    
+    self.wasNewSelection = button.tag != self.selectedSegmentIndex;
```

Comment:
Do you think we should move this logic into the same conditional checking for `!self.isTransitioning`? It seems that this library only mutates its state provided it is not transitioning